tests: make yarn unit run all available unit tests

brendankenny opened the pull request:
we keep growing `yarn unit` by adding every new `yarn unit-*`, but `jest.config.js` has to have references to these files anyways, so `yarn jest` works just fine if you just want to run all the unit tests :)

This also fixes some overlaps and cracks between the `yarn unit-*`s. The current `yarn unit` doesn't run
- ~`clients/test/extension/popup-test.js`~
- `clients/test/extension/settings-controller-test.js`
- `clients/test/lightrider-entry-test.js`
- ~`docs/recipes/integration-test/example-lh-auth.test.js`~
- `third-party/chromium-synchronization/inspector-issueAdded-types-test.js`
- `third-party/chromium-synchronization/installability-errors-test.js`

the last two might be on purpose but seems like they should be in the testing path if the goal is for changes in Canary to alert us to needed updates? It also made us miss that the path to `LH_ROOT` was wrong in `inspector-issueAdded-types-test.js` and it doesn't run at all.

We're also running the `flow-report` tests twice because they're picked up by both `yarn unit-report` and `yarn unit-flow`.

This is most important in CI, where unit.yml runs them via [`yarn unit:ci`](https://github.com/GoogleChrome/lighthouse/blob/35d5b214386ea31cdf814095e477e747c118c0bf/.github/workflows/unit.yml#L55) and should be running all the tests.

brendankenny commented:
> * `clients/test/extension/popup-test.js`
> * `clients/test/extension/settings-controller-test.js`
> * `clients/test/lightrider-entry-test.js`
> * `docs/recipes/integration-test/example-lh-auth.test.js`

well, these _are_ covered by `yarn test-clients` and `yarn test-docs` in `ci.yml`. I'm pretty fine running them with `yarn unit` if they're going to be listed in the `jest.config.js`, though.

Context here is that https://github.com/GoogleChrome/lighthouse/pull/13146 is adding yet another directory with tests in it, and I don't really have any interest in adding a `yarn unit-shared` with bonus `&& yarn unit-shared` and `&& npm run unit-shared -- --ci` when `yarn unit` could just run them all and `yarn jest shared` works just fine for narrowing to just those tests already :)

adamraine commented on .github/workflows/unit.yml:
```diff
@@ -44,7 +44,7 @@ jobs:
         pip install protobuf==3.7.1
 
     - run: yarn install --frozen-lockfile --network-timeout 1000000
-    - run: yarn build-report
```
What were we missing out on before this change?

brendankenny replied:
> What were we missing out on before this change?

tests under `clients/test/extension/` needed `yarn build-extension` and `lightrider-entry-test.js` needed `yarn build-lr` and with `yarn build-report` already needed, `build-all` just seemed easier :)

brendankenny commented:
rather than make `docs/recipes/integration-test/example-lh-auth.test.js` work in the unit tests, would rather just remove it from the unit tests (#13150)

brendankenny commented:
updated with docs test changes from #13150 so `yarn test-docs` doesn't run under `yarn unit`.

Also moved the `*-test-pptr.js` tests previously identified as flaky ([they have automatic retry in CI](https://github.com/GoogleChrome/lighthouse/blob/e6921d9b1980ad8b50a3ad1705ff1e2102487140/.github/workflows/ci.yml#L36-L43)) out of the default unit tests. They now only run under their respective `yarn test-clients`/`yarn test-viewer`/`yarn test-treemap` and not under `yarn unit`, which is exactly the current behavior.

`fraggle-rock/**/*-test-pptr.js` has been running in `unit.yml` without flakes for a while now, so seems ok to leave under unit tests for now.

Mbellsudteen commented:
ในวันที่ ศ. 1 ต.ค. 2021 04:19 น. devtools-bot ***@***.***>
เขียนว่า:

> Merged #13148 <https://github.com/GoogleChrome/lighthouse/pull/13148>
> into master.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleChrome/lighthouse/pull/13148#event-5390426276>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZGLUSXV7HOQW76I2VY4W3UETH47ANCNFSM5FBDOQ7Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
